src/boosting/score_updater.hpp

```diff
@@ -18,12 +18,12 @@ class ScoreUpdater {
   * \brief Constructor, will pass a const pointer of dataset
   * \param data This class will bind with this data set
   */
-  explicit ScoreUpdater(const Dataset* data)
-    :data_(data) {
+  explicit ScoreUpdater(const Dataset* data, int num_class)
+    :data_(data), num_class_(num_class) {
```

Comment:
Should we really need to pass "num_class" to score_updater?
I think we can add an parameter "offset" in every "AddScore". 
And pass "num_data_*num_class_" as size in constructor